### PR TITLE
Add new trusted intermediary groups.

### PR DESCRIPTION
**Pull request by ndmckinley**
### Description
Adds a form to the bottom of the page that lets you add a new TI group.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

Screenshot: 
Screen Shot 2021-04-30 at 12 29 10 PM

### Issue(s)
Related to #251.

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java [hunk 1]
```diff
@@ -20,7 +22,18 @@ public TrustedIntermediaryManagementController(
...
+  public Result index(Http.Request request) {
+    return ok(listView.render(userRepository.listTrustedIntermediaryGroups(), request));
+  }
+
+  @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
+  public Result newGroup(Http.Request request) {
+    String[] name = request.body().asFormUrlEncoded().get("name");
+    String[] description = request.body().asFormUrlEncoded().get("description");
+    Preconditions.checkState(name.length == 1, "Need exactly one `name` form entry.");
+    Preconditions.checkState(description.length == 1, "Need exactly one `name` form entry.");
```
Comment (natsid) [inline universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java:34]: It looks like `Preconditions.checkState` throws an exception if the requirements aren't met, which I don't think we want to do here. I think we want to return a friendly validation error message and re-render the form instead.

The standard way we've been handling forms/POSTs is via `Form` classes. See [BlockForm](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/forms/BlockForm.java) for a simple 2-field example. And see how it's used in [AdminProgramBlocksController](https://github.com/seattle-uat/civiform/blob/b7925499481906412758407e9b661cebe1f02bfc/universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlocksController.java#L99).

I think creating a `TrustedIntermediaryGroupForm` might help streamline the code to re-render the form with the values the user (admin) originally provided. Let me know if I can help.

**Comment by natsid** [comment]
Hey @ndmckinley - just left a comment about the form request handling. I was wondering if you could also attach a screenshot of the new form to the PR description so I can get a visual 😄 .

File: universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java [hunk 2]
```diff
@@ -20,7 +22,18 @@ public TrustedIntermediaryManagementController(
...
 
   @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
-  public Result index() {
-    return ok(listView.render(userRepository.listTrustedIntermediaryGroups()));
+  public Result index(Http.Request request) {
+    return ok(listView.render(userRepository.listTrustedIntermediaryGroups(), request));
+  }
+
+  @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
+  public Result newGroup(Http.Request request) {
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java:40]: (blocking)  this method be called `create` to match all the other controllers

**Review by kendrickt — APPROVED** [review]
a couple naming things. + natalie's form comment. 

Otherwise, lgtm.

**Comment by kendrickt** [comment]
Could you add a screenshot to this PR?

**Comment by ndmckinley** [comment]
Added a screenshot.

File: universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java (hunk 2 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/app/controllers/admin/TrustedIntermediaryManagementController.java]: Done!

**Pull request merged**